src/LuauExt.cpp

```diff
@@ -428,6 +429,39 @@ lsp::Diagnostic createParseErrorDiagnostic(const Luau::ParseError& error, const
     return diagnostic;
 }
 
+// TODO: Dirty hack for invalid definitionModuleName on type aliases for solver v2!
+// Remove after https://github.com/luau-lang/luau/issues/1441 is closed!
+std::optional<Luau::ModuleName> lookupTypeDefinitionModule(Luau::TypeId type)
+{
+    if (!FFlag::LuauSolverV2)
+    {
+        return Luau::getDefinitionModuleName(type);
+    }
+
+    type = Luau::follow(type);
+    if (auto ttv = Luau::get<Luau::TableType>(type); ttv && ttv->definitionModuleName.empty())
+    {
+        if (type->owningArena && type->owningArena->owningModule)
+            return type->owningArena->owningModule->name;
```

Comment:
This seems incredibly hacky. Are we sure owningModule is the correct module in all cases? If a type got copied out of a module, the owning module would change

Reply:
iirc my brief testing had it correct for even imported types from other modules. the problem is I couldn't find any other way to get this information given the luau bug making definitionModuleName empty.

Reply:
https://github.com/luau-lang/luau/pull/1495
new solution : fix the luau bug